File: Selenium/Activities/Activity16.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.select import Select
with webdriver.Firefox() as driver:
    wait=WebDriverWait(driver,timeout=10)
    driver.get("https://training-support.net/webelements/selects")
    print("the title of the page is: " ,driver.title)

    dropdown = driver.find_element(By.CSS_SELECTOR,"select.h-10")
    singleSelect = Select(dropdown)
    singleSelect.select_by_visible_text("Two")
    print("Second option: " , singleSelect.first_selected_option.text)
    singleSelect.select_by_index(3)
    print("third option: ",singleSelect.first_selected_option.text)
    singleSelect.select_by_value("four")
    print("fourth option: ",singleSelect.first_selected_option.text)
    
    all_options = singleSelect.options
    print("options in the dropdown: ")
    for option in all_options:
        print(option.text)

    driver.quit()

File: Selenium/Activities/Activity16.java
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.Select;

public class Activity16 {
	public static void main(String args[]) {
		WebDriver driver = new FirefoxDriver();
		driver.get("https://training-support.net/webelements/selects");
		System.out.println("the title of the page is: "+driver.getTitle());
		WebElement dropdown = driver.findElement(By.cssSelector("select.h-10"));
		Select singleSelect = new Select(dropdown);
		
		singleSelect.selectByVisibleText("Two");
		System.out.println("Second option: " + singleSelect.getFirstSelectedOption().getText());
		singleSelect.selectByIndex(3);
		System.out.println("third option: " + singleSelect.getFirstSelectedOption().getText());
		singleSelect.selectByValue("four");
		System.out.println("fourth option: "+singleSelect.getFirstSelectedOption().getText());
		
		List<WebElement> all_options = singleSelect.getOptions();
		System.out.println("options in the dropdown: ");
		for(WebElement option:all_options) {
			System.out.println(option.getText());
		}
		
		driver.quit();
		
	}

}
